File: MacApp/Bridge/Models/ScanProgress.swift
import Foundation

/// Progress update from the Go scan engine.
struct ScanProgress: Codable {
    let currentIP: String
    let percent: Double
    let activeThreads: Int32
    let state: String

    private enum CodingKeys: String, CodingKey {
        case currentIP = "current_ip"
        case percent
        case activeThreads = "active_threads"
        case state
    }
}

/// Statistics from the Go scan engine.
struct ScanStats: Codable {
    let total: Int
    let alive: Int
    let withPorts: Int
}

File: WindowsApp/GoNetworkScanner/Bridge/Models/ScanProgress.cs
using System.Text.Json.Serialization;

namespace GoNetworkScanner.Bridge.Models;

public class ScanProgress
{
    [JsonPropertyName("current_ip")]
    public string CurrentIP { get; set; } = "";

    [JsonPropertyName("percent")]
    public double Percent { get; set; }

    [JsonPropertyName("active_threads")]
    public int ActiveThreads { get; set; }

    [JsonPropertyName("state")]
    public string State { get; set; } = "";
}

public class ScanStats
{
    [JsonPropertyName("total")]
    public int Total { get; set; }

    [JsonPropertyName("alive")]
    public int Alive { get; set; }

    [JsonPropertyName("withPorts")]
    public int WithPorts { get; set; }
}
